report-api/src/main/java/no/sikt/nva/data/report/api/fetch/xlsx/Excel.java

```diff
@@ -1,56 +1,67 @@
 package no.sikt.nva.data.report.api.fetch.xlsx;
 
+import java.io.ByteArrayOutputStream;
 import java.io.IOException;
 import java.io.OutputStream;
 import java.util.List;
-import org.apache.poi.ss.usermodel.CellStyle;
-import org.apache.poi.ss.usermodel.FillPatternType;
-import org.apache.poi.ss.usermodel.IndexedColors;
-import org.apache.poi.ss.usermodel.Sheet;
+import org.apache.poi.ss.usermodel.Row;
 import org.apache.poi.ss.usermodel.Workbook;
 import org.apache.poi.xssf.usermodel.XSSFWorkbook;
 
 public record Excel(Workbook workbook) {
 
     public static Excel fromJava(List<String> headers, List<List<String>> data) {
-        var excel = new Excel(new XSSFWorkbook());
-        var sheet = excel.workbook.createSheet();
-        addHeadersToSheet(headers, sheet);
-        addDataToSheet(data, sheet);
+        var excel = new Excel(createWorkbookWithOneSheet());
+        excel.addHeaders(headers);
+        excel.addData(data);
         return excel;
     }
 
-    public void write(OutputStream outputStream) throws IOException {
-        workbook.write(outputStream);
-        workbook.close();
+    public static Excel errorReport() {
+        var excel = new Excel(createWorkbookWithOneSheet());
+        excel.addData(List.of(List.of("Unexpected error occurred. Please contact support.")));
+        return excel;
     }
 
-    private static void addHeadersToSheet(List<String> headers, Sheet sheet) {
-        var header = sheet.createRow(0);
-        var headerStyle = getCellStyle(sheet);
-
-        for (var headerCounter = 0; headerCounter < headers.size(); headerCounter++) {
-            var headerCell = header.createCell(headerCounter);
-            headerCell.setCellStyle(headerStyle);
-            headerCell.setCellValue(headers.get(headerCounter));
+    public void addData(List<List<String>> data) {
+        var sheet = workbook.getSheetAt(0);
+        for (List<String> cells : data) {
+            var nextRow = sheet.getLastRowNum() + 1;
+            addCells(sheet.createRow(nextRow), cells);
         }
     }
 
-    private static CellStyle getCellStyle(Sheet sheet) {
-        var headerStyle = sheet.getWorkbook().createCellStyle();
-        headerStyle.setFillForegroundColor(IndexedColors.GREY_25_PERCENT.getIndex());
-        headerStyle.setFillPattern(FillPatternType.SOLID_FOREGROUND);
-        return headerStyle;
+    public byte[] toBytes() {
+        var byteArrayOutputStream = new ByteArrayOutputStream();
+        try {
+            this.write(byteArrayOutputStream);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+        return byteArrayOutputStream.toByteArray();
```

Comment:
Can this line be placed in the try block? It likely has no real effect, but perhaps cleaner.